UnityAds/UnityRouter.m

```diff
@@ -50,6 +54,17 @@ + (UnityRouter *)sharedRouter
 }
 
 - (void)initializeWithGameId:(NSString *)gameId withCompletionHandler:(void (^)(NSError *))complete {
+    if ([UnityAds isInitialized]) {
+        if (complete != nil) {
+            complete(nil);
+        }
+        return;
+    }
+    // If this method was called multiple times from different threads, we want to call all completion handlers once initialization is done.
+    if (complete != nil) {
+        [self.completionBlocks addObject:complete];
+    }
```

Comment:
should we wrap it into synchronize? and the same for callCompletions?

Reply:
good catch, fixed